jsk_perception/node_scripts/ssd_object_detector.py

```diff
@@ -0,0 +1,176 @@
+        return label_names
+
+    def config_callback(self, config, level):
+        self.model.nms_thresh = config.nms_thresh
+        self.model.score_thresh = config.score_thresh
+        self.profiling = config.profiling
+        return config
+
+    def image_cb(self, msg):
+        if self.profiling:
+            rospy.loginfo("callback start: incomming msg is %s msec behind" % ((rospy.Time.now() - msg.header.stamp).to_sec() * 1000.0))
+        tprev = time.time()
+        try:
+            # transform image to RGB, float, CHW
+            img = self.cv_bridge.imgmsg_to_cv2(msg, desired_encoding="rgb8")
+            img = np.asarray(img, dtype=np.float32)
+            img = img.transpose(2, 0, 1)  # (H, W, C) -> (C, H, W)
+        except Exception as e:
+            rospy.logerr("Failed to convert image: %s" % str(e))
+            return
+        if self.profiling:
+            tcur = time.time()
+            rospy.loginfo("%s: elapsed %f msec" % ("convert", (tcur-tprev)*1000))
+            tprev = tcur
+
+        bboxes, labels, scores = self.model.predict([img])
+        bboxes, labels, scores = bboxes[0], labels[0], scores[0]
+
+        if self.profiling:
+            tcur = time.time()
+            rospy.loginfo("%s: elapsed %f msec" % ("predict", (tcur-tprev)*1000))
+            tprev = tcur
+
+        rect_msg = RectArray(header=msg.header)
+        for bbox in bboxes:
+            rect = Rect(x=bbox[1], y=bbox[0],
+                        width= bbox[3] - bbox[1],
+                        height=bbox[2] - bbox[0])
+            rect_msg.rects.append(rect)
+
+        if self.profiling:
+            tcur = time.time()
+            rospy.loginfo("%s: elapsed %f msec" % ("make rect msg", (tcur-tprev)*1000))
+            tprev = tcur
+
+        cls_msg = ClassificationResult(
+            header=msg.header,
+            classifier=self.classifier_name,
+            target_names=self.label_names,
+            labels=labels,
+            label_names=[self.label_names[l] for l in labels],
+            label_proba=scores,
+        )
+
+        if self.profiling:
+            tcur = time.time()
+            rospy.loginfo("%s: elapsed %f msec" % ("make cls msg", (tcur-tprev)*1000))
+            tprev = tcur
+
+        self.pub_rects.publish(rect_msg)
+        self.pub_class.publish(cls_msg)
+
+        if self.profiling:
+            tcur = time.time()
+            rospy.loginfo("%s: elapsed %f msec" % ("publish msg", (tcur-tprev)*1000))
+            tprev = tcur
+
+        if self.visualize:
+            self.publish_bbox_image(img, bboxes, labels, scores)
+
+        if self.profiling:
+            tcur = time.time()
+            rospy.loginfo("%s: elapsed %f msec" % ("callback end", (tcur-tprev)*1000))
+            tprev = tcur
+
+    def publish_bbox_image(self, img, bbox, label, score):
+        vis_bbox(img, bbox, label, score,
+                 label_names=self.label_names)
+        fig = plt.gcf()
```

Comment:
i think we do not need to plot in this node.

Reply:
@knorth55 Why?

Reply:
faster_rcnn nodes also draw boxes and labels, so I think we can make a common node like `draw_boxes`

Reply:
@knorth55 Thanks! I created pull request to generalize visualization part. https://github.com/jsk-ros-pkg/jsk_recognition/pull/2205

Reply:
- #2205 was merged.
- Updated to use `draw_rects` for visualization
- Remove visualization function from `ssd_object_detector.py` node
- Updated docs